tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionRewriteSuite.scala

```diff
@@ -67,8 +67,27 @@ class RegularExpressionRewriteSuite extends AnyFunSuite {
       PrefixRange("火花急流", 1, 19968, 40869),
       NoOptimization, // starts with PrefixRange not supported
       NoOptimization, // starts with PrefixRange not supported
-      PrefixRange("", 6, 48, 57),
-      PrefixRange("", 3, 48, 57)
+      NoOptimization, // .* can't match line break so can't be optimized
+      NoOptimization  // .* can't match line break so can't be optimized
+    )
+    verifyRewritePattern(patterns, excepted)
+  }
+
+  test("regex rewrite multiple contains") {
+    import RegexOptimizationType._
+    val patterns = Seq(
+      "(abc|def).*",
+      ".*(abc|def|ghi).*",
+      "((abc)|(def))",
+      "(abc)|(def)",
+      "(火花|急流)"
+    )
+    val excepted = Seq(
+      MultipleContains(Seq("abc", "def")),
+      MultipleContains(Seq("abc", "def", "ghi")),
+      MultipleContains(Seq("abc", "def")),
+      NoOptimization,
```

Comment:
Just curious why does `((abc)|(def))` can be optimized but `(abc)|(def)` can not?

Reply:
An orthogonal high-level question: have we considered making these regex rewrites in libcudf such that other libcudf users benefit as well.

Reply:
> An orthogonal high-level question: have we considered making these regex rewrites in libcudf such that other libcudf users benefit as well.

Yes, a better way is to speed up the cudf regex engine itself, https://github.com/rapidsai/cudf/issues/15611 proposed to do similar optimization, but rewriting it from Spark side are some lower hanging fruits to take.

Reply:
> Just curious why does `((abc)|(def))` can be optimized but `(abc)|(def)` can not?

Good question, this is actually a bug:

In `GpuRLikeMeta` I called `matchSimplePattern` with `regexAst.children()` and assumed that most of the regexAst is a `RegexSequence`, but this is not always true. 

For example `(abc)|(def)` is a single `RegexChoice(RegexSequence(ListBuffer(RegexChar(a), RegexChar(b), RegexChar(c))), RegexSequence(ListBuffer(RegexChar(d), RegexChar(e), RegexChar(f))))`, so I pass the children of it and it failed to be optimized.

Fixed it.